cart/src/main/java/com/yas/cart/repository/CartItemV2Repository.java

```diff
@@ -15,7 +15,12 @@ public interface CartItemV2Repository extends JpaRepository<CartItemV2, CartItem
     @Lock(LockModeType.PESSIMISTIC_WRITE)
     @QueryHints({@QueryHint(name = "jakarta.persistence.lock.timeout", value = "0")})
     @Query("SELECT c FROM CartItemV2 c WHERE c.customerId = :customerId AND c.productId = :productId")
-    Optional<CartItemV2> findWithLock(String customerId, Long productId);
+    Optional<CartItemV2> findOneWithLock(String customerId, Long productId);
```

Comment:
```suggestion
    Optional<CartItemV2> findByCustomerIdAndProductId(String customerId, Long productId);
```

Reply:
Thank you for your suggestion. I believe the inclusion of `WithLock` clarifies the intent that these methods are designed to lock records, which is important for other developers to understand their behavior. 

However, I'm open to discussing this further if you have additional thoughts.

Reply:
You can add java doc to this method, it will make it easier for other developers to find the pessimistic lock.

Reply:
I've just pushed the changes. Could you please take a look at them when you have a moment? Thank you for your feedback.